examples/pyo3-pytests/build.rs

```diff
@@ -0,0 +1,28 @@
+use std::process::Command;
+
+fn main() {
+    let out = Command::new("python")
+        .args(&["-c", "import sys; import platform; print(sys.version_info[1]); print(platform.python_implementation())"])
+        .output()
+        .expect("python version did not print");
+
+    let output = String::from_utf8_lossy(&out.stdout);
+    let mut lines = output.trim().lines();
+
+    println!("{}", output);
+
+    let version: u8 = lines
+        .next()
+        .unwrap()
+        .parse()
+        .expect("python version was not parsed");
+    let implementation = lines.next().unwrap();
+
+    for each in 6..version {
+        println!("cargo:rustc-cfg=Py_3_{}", each);
+    }
+
+    if implementation == "PyPy" {
+        println!("cargo:rustc-cfg=PyPy");
+    }
```

Comment:
Why do we need this?

Reply:
This used to be [in setup.py](https://github.com/PyO3/pyo3/blob/1751a7fb19b47a7dcc1f74d1ed442d74e4f39dbc/examples/rustapi_module/setup.py#L8), but we now build the tests with maturin instead of setuptools-rust.

We need it because some tests only compile on certain python / PyPy combinations.

IMO this `build.rs` method is better anyway because it would still work if we changed the examples back to `setuptools_rust`.

It's a little ugly to have to call a Python interpreter here. I have an idea how to split some of pyo3's `build.rs` logic into a separate crate `pyo3-build-config,` which this build.rs could then re-use. I've been torn for a while on whether that refactoring is worth the additional complexity; maybe I should hurry up and push that PR for a while so that others can see what they think.

Reply:
I'm going to move this into a new issue so that it doesn't block this PR with all the new examples, which I think has value even if there's this this wart.